sim/battle-actions.ts

```diff
@@ -528,14 +528,22 @@ export class BattleActions {
 			!(move.hasSheerForce && pokemon.hasAbility('sheerforce')) &&
 			!move.flags['futuremove']
 		) {
-			const originalHp = pokemon.hp;
+			let originalHp = pokemon.hp;
 			this.battle.singleEvent('AfterMoveSecondarySelf', move, null, pokemon, target, move);
 			this.battle.runEvent('AfterMoveSecondarySelf', pokemon, target, move);
```

Comment:
I don't know how the other per-move `AfterMoveSecondarySelf` handlers should resolve priority compared to other `AfterMoveSecondarySelf` effects, but in case it helps, you can combine them in one priority system as follows:
```suggestion
			this.battle.runEvent('AfterMoveSecondarySelf', pokemon, target, move, null, true);
```

Reply:
That code isn't mine, I prefer not to touch it.